### PR TITLE
Reactions

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds a new voting system, names-attached reactions.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204491047346355) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
I just wrote this up as a possible project (inspired by shared conversations I think), and then look what showed up like an hour later.

File: packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+  function toggleReaction(name: string) {
+    if (!currentUser) {
+      openLoginDialog();
+      return;
+    }
+    
+    if (getCurrentUserReactionVote(name)) {
+      clearCurrentUserReaction(name);
+    } else {
+      const initialVote = "created"; //TODO: "created" vs "seconded"
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx:174]: This has a TODO - not sure if still relevant?

File: packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx:346]: This has a TODO - not sure if still relevant?

File: packages/lesswrong/lib/voting/reactions.tsx [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+ *
+ * When you add a downloaded image, record where you got it in
+ * `public/reactionImages/sources.txt`. This is required to be able to meet the
+ * attribution requirement of many licenses. If you get an image from a source
+ * that hasn't been used before, you have to check its license and make sure
+ * it's suitable.
+ */
+
+// OpenClipart and OpenMoji
+/*export const namesAttachedReactions: NamesAttachedReactionType[] = [
```
Comment (b0b3rt) [inline packages/lesswrong/lib/voting/reactions.tsx:72]: Does this need to be cleaned up?  There's a TODO at the bottom

**Review by b0b3rt — APPROVED** [review]

File: packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/votes/NamesAttachedReactionsVoteOnComment.tsx:174]: Tracking who created vs who seconded is a nice-to-have but not essential (particularly not for the beta-test open thread). So, leaving in the TODO for now.

File: packages/lesswrong/lib/voting/reactions.tsx (hunk 3 above)
Comment (jimrandomh) [reply packages/lesswrong/lib/voting/reactions.tsx]: This is a previous palette in a different aesthetic, which became cruft when I went to Noun-Project-only. Removed.

**Pull request merged**